client/driver/docker.go

```diff
@@ -153,8 +153,14 @@ func createContainer(ctx *ExecContext, task *structs.Task, logger *log.Logger) d
 		hostConfig.PortBindings = dockerPorts
 	}
 
+	// Merge Nomad-native with user-specified envvars
+	env := TaskEnvironmentVariables(ctx, task).List()
```

Comment:
How about adding a SetEnv method to TaskEnvironment and then call it from TaskEnvironmentVariables alongside the call to SetMeta? That would give the functionality to all the drivers.